### DISCUSSION TITLE
Filament Forms DatePicker accessibility

### DISCUSSION THREAD

**Discussion by wik0rl** [filamentphp/filament#4903] [Help]
Hallo everbody,

I developed a desk reservation app for my company and using fillament forms for it. Today I got the information that the DatePicker form field is not readable by [(https://www.nvaccess.org/download/)]. The user who is using NVDA is blind and has no other possibility to use my software.

Do I have any chance fixing the problem?

Thank you in advance!

Best regards
Michael

**Comment by zepfietje** [comment]
Hey @wik0rl, thanks for reporting!

Could you point us at the exact issues?
Only then we'll be able to fix the problem.

**Reply by wik0rl** [reply]
I have the following form. When the specific user tries to access the "start_date" field with the screen reader, it reads the dropdown as a table and therefore isn't able to select a date. One solution would be to make the DatePicker input field writeable like a normal input field.

```
protected function getFormSchema(): array
{
      return [
            Section::make('Datum')
                ->description('Wähle ein Datum für die Reservierung')
                ->schema([
                        DatePicker::make('start_date')
                            ->minDate(now()->startOfDay())
                            ->displayFormat('d.m.Y')
                            ->required()
                            ->columnSpan(2),
                        Toggle::make('reccurance')
                            ->label('Wiederholende Reservierung?')
                            ->reactive(),

            ])->columns([
                'sm' => 1,
                'xl' => 2]),
        ];
}
```

**Reply by danharrin** [reply]
AFAIK I looked at W3 accessibility standards when I built the date picker. Have you reviewed those? Please let me know specifically which standards we aren't following.

Alternatively, you can use the native HTML date picker with `TextInput::make()->type('date')`
